plugins/module_utils/network/nxos/facts/legacy/base.py

```diff
@@ -342,20 +342,23 @@ def populate_structured_ipv6_interfaces(self, data):
 
     def populate_structured_neighbors_lldp(self, data):
         objects = dict()
-        data = data["TABLE_nbor"]["ROW_nbor"]
+        try:
+            data = data["TABLE_nbor"]["ROW_nbor"]
```

Comment:
Thinking out loud - if we restrict the try-except to this part _only_, it might be more accurate.

If there's a KeyError in the code below, it needs to fail the task and not mask it. Wdyt?

Reply:
If it enters the `except`, we return an empty dict.